### PR TITLE
WT-11787 Cut WiredTiger 11.2.0 release

### PR DESCRIPTION
**Pull request by lukech**
Cut WiredTiger 11.2.0 release by following [this](https://github.com/wiredtiger/wiredtiger/wiki/HOWTO:BuildARelease) procedure (up to step 6).

Note: the compatibility test update (step 5) was made last time the version was bumped to 11.2.0 in the `RELEASE_INFO` file (WT-10135).

### CODE REVIEW

File: cmake/configs/version.cmake [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# Generated by dist/s_version. Do not modify.
-set(WT_VERSION_MAJOR 11)
-set(WT_VERSION_MINOR 2)
-set(WT_VERSION_PATCH 0)
-set(WT_VERSION_STRING "WiredTiger 11.2.0: (November 10, 2022)")
+    # Generated by dist/s_version. Do not modify.
```
Comment (etienneptl) [inline cmake/configs/version.cmake:1]: I don't think we should have those spaces here. I will create a ticket to remove those, you should regenerate the release after this is done.

File: src/docs/arch-glossary.dox [hunk 2]
```diff
@@ -103,8 +103,8 @@ they are used in other settings.
 <tr><td>table<td>schema<td>an abstraction that allows one or more btrees to be used in a coordinated way, which is specified by the schema.  The table coordinates indices and column groups.
 <tr><td>thread<td>general<td>a thread of control.  WiredTiger uses pthreads for POSIX systems.
 <tr><td>thread group<td>general<td>a group of threads that perform a common task.  Eviction is performed by a thread group.
-<tr><td>time window<td>durable history<td>a set of start and stop transaction ids and timestamps which determine when an update is visible.
 <tr><td>timestamp<td>transactions<td>refers to one of several types of 64 bit values used to control transactional visibility of data in WiredTiger.  See oldest timestamp, stable timestamp, commit timestamp, read timestamp.
+<tr><td>time window<td>durable history<td>a set of start and stop transaction ids and timestamps which determine when an update is visible.
```
Comment (etienneptl) [inline src/docs/arch-glossary.dox:107]: I will create a ticket for this, it should not be fixed here.

File: cmake/configs/version.cmake (hunk 1 above)
Comment (etienneptl) [reply cmake/configs/version.cmake:1]: We should have had tabs before but somehow we don't. I see that WT-10302 changed the tabs to spaces.

Comment (lukech) [reply cmake/configs/version.cmake:1]: Good catch. This change was generated by the `s_all -f` command. The tabs->spaces change might have missed this path.

File: src/docs/arch-glossary.dox (hunk 2 above)
Comment (lukech) [reply src/docs/arch-glossary.dox:107]: Thanks, that would be great.
It's worth noting in the PR build there's another related error (s_all does not like the change made here?)
```
[2023/10/06 01:25:39.588] ####################### MESSAGE ############################
[2023/10/06 01:25:39.588] + echo s_all run of: "./s_docs" resulted in:
[2023/10/06 01:25:39.588] s_all run of: "./s_docs" resulted in:
[2023/10/06 01:25:39.588] + sed -e s/^/    / __s_all_tmp_s_docs
[2023/10/06 01:25:39.589]     89d88
[2023/10/06 01:25:39.589]     < timestamp
[2023/10/06 01:25:39.589]     90a90
[2023/10/06 01:25:39.589]     > timestamp
[2023/10/06 01:25:39.589]     =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[2023/10/06 01:25:39.589]     arch-glossary.dox entries out of order
[2023/10/06 01:25:39.589]     =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
```

Comment (lukech) [reply src/docs/arch-glossary.dox]: Reverting the line-shifting change made the above error go away. It's not yet clear unfortunately what could have caused the difference between my local dev env and Evergreen (both running sort version 8.30).

**Review by korteland — APPROVED** [review]
LGTM! Assuming the generated docs look OK.

File: src/docs/upgrading.dox [hunk 3]
```diff
@@ -1,5 +1,34 @@
 /*! @page upgrading Upgrading WiredTiger applications
 
+@section version_1120 Upgrading to Version 11.2.0
+<dl>
+
+<dt>Connection config "operation_timeout_ms"</dt>
```
Comment (etienneptl) [inline src/docs/upgrading.dox:6]: You use `config` here and `configuration` line 18, we should be consistent. I prefer the full word as it is a user-facing doc.

File: src/docs/upgrading.dox [hunk 4]
```diff
@@ -1,5 +1,34 @@
...
+<dt>Cursor configuration "prefix_search"</dt>
+<dd>
+The cursor configuration "prefix_search" is now deprecated. The option is still accepted
+in this version but will have no effect when set. Users should look into the new bound
+cursor interface.
+</dd>
+
+<dt>Session API "flush_tier" function</dt>
+<dd>
+The session API "flush_tier" function is removed. Users of this function should move to
```
Comment (etienneptl) [inline src/docs/upgrading.dox:27]: @ddanderson, could you please verify this statement to make sure we understood the change correctly?

**Review by etienneptl — COMMENTED** [review]
Left a comment and a question. It would be great to verify if the automated script would get to that stage too.

File: src/docs/upgrading.dox (hunk 3 above)
Comment (lukech) [reply src/docs/upgrading.dox]: Good catch. Fixed in [9b744b1](https://github.com/wiredtiger/wiredtiger/pull/9724/commits/9b744b1dd34097de332f27237d6ab56683a0af6b)

File: src/docs/upgrading.dox (hunk 4 above)
Comment (ddanderson) [reply src/docs/upgrading.dox:27]: @etienneptl, yes you have it correct.

**Comment by lukech** [comment]
@korteland @etienneptl I added the missing 11.1.0 upgrading entries and a link to the "10.0.0 upgrading doc" for older versions. See [2dc5145](https://github.com/wiredtiger/wiredtiger/pull/9724/commits/2dc5145f548cc99dbfe80f68e2d42486c26b68cb)

**Comment by korteland** [comment]
Thanks @lukech - new changes also look good.

**Review by etienneptl — APPROVED** [review]
LGTM so far!

**Pull request merged**